source/common/network/io_socket_handle_impl.cc

```diff
@@ -84,8 +84,14 @@ Api::IoCallUint64Result IoSocketHandleImpl::readv(uint64_t max_length, Buffer::R
     num_bytes_to_read += slice_length;
   }
   ASSERT(num_bytes_to_read <= max_length);
-  return sysCallResultToIoCallResult(Api::OsSysCallsSingleton::get().readv(
+  auto result = sysCallResultToIoCallResult(Api::OsSysCallsSingleton::get().readv(
       fd_, iov.begin(), static_cast<int>(num_slices_to_read)));
+
+  // Some tests try to read without initializing the file_event.
```

Comment:
Are these test bugs that need to be fixed?

Reply:
Not really just convenience for tests. Consider a test that creates an io handle to do a single read/write, this test might not care about the file_event.